File: go-ut-bench/datasets/go/go_code_files_self_contained/interface_mock/interface_mock_001.go
package main

type Logger interface {
	Log(message string)
	GetLogs() []string
}

type ConsoleLogger struct {
	logs []string
}

func NewConsoleLogger() *ConsoleLogger {
	return &ConsoleLogger{logs: []string{}}
}

func (cl *ConsoleLogger) Log(message string) {
	cl.logs = append(cl.logs, message)
}

func (cl *ConsoleLogger) GetLogs() []string {
	result := make([]string, len(cl.logs))
	for i, log := range cl.logs {
		result[i] = log
	}
	return result
}

type FileLogger struct {
	filename string
	logs     []string
}

func NewFileLogger(filename string) *FileLogger {
	return &FileLogger{filename: filename, logs: []string{}}
}

func (fl *FileLogger) Log(message string) {
	fl.logs = append(fl.logs, "["+fl.filename+"] "+message)
}

func (fl *FileLogger) GetLogs() []string {
	result := make([]string, len(fl.logs))
	for i, log := range fl.logs {
		result[i] = log
	}
	return result
}

type Application struct {
	logger Logger
}

func NewApplication(logger Logger) *Application {
	return &Application{logger: logger}
}

func (app *Application) Run(taskName string) string {
	app.logger.Log("Starting task: " + taskName)
	result := "Task " + taskName + " completed"
	app.logger.Log(result)
	return result
}

func (app *Application) GetHistory() []string {
	return app.logger.GetLogs()
}

File: go-ut-bench/datasets/python/python_code_files_self_contained/interface_mock/interface_mock_001.py
from abc import ABC, abstractmethod


class Logger(ABC):
    @abstractmethod
    def log(self, message: str) -> None:
        pass
    
    @abstractmethod
    def get_logs(self) -> list:
        pass


class ConsoleLogger(Logger):
    def __init__(self):
        self._logs = []
    
    def log(self, message: str) -> None:
        self._logs.append(message)
    
    def get_logs(self) -> list:
        return self._logs.copy()


class FileLogger(Logger):
    def __init__(self, filename: str):
        self._filename = filename
        self._logs = []
    
    def log(self, message: str) -> None:
        self._logs.append(f"[{self._filename}] {message}")
    
    def get_logs(self) -> list:
        return self._logs.copy()


class Application:
    def __init__(self, logger: Logger):
        self._logger = logger
    
    def run(self, task_name: str) -> str:
        self._logger.log(f"Starting task: {task_name}")
        result = f"Task {task_name} completed"
        self._logger.log(result)
        return result
    
    def get_history(self) -> list:
        return self._logger.get_logs()